app/tools/android/test_data/extensions/contactextension/src/com/example/extension/ContactExtension.java

```diff
@@ -88,16 +88,13 @@ String readContact(String name) {
                     ContactsContract.Contacts.DISPLAY_NAME + "='" + name + "'",
                     null, null);
 
-            if (phones.moveToNext()) {
-                String phone = phones.getString(phones.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER));
-                phones.close();
-                cursor.close();
-                return phone;
-            } else {
-                Log.w(TAG, "no phone number for " + name);
-                cursor.close();
-                return "";
+            String phone = "";
+            if (phones.getCount() > 0) {
```

Comment:
The cursor returned from getContentResolver().query may be null but no null check here.